pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -698,13 +697,6 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 		if corev1.PodSucceeded != pod.Status.Phase {
 			if corev1.PodFailed == pod.Status.Phase {
 				c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPodFailed, "Populator failed: %s", pod.Status.Message)
-				// Delete failed pods so we can try again
-				if !plan.RestartLimitExceeded(pod) {
-					err = c.kubeClient.CoreV1().Pods(populatorNamespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
-					if err != nil {
-						return err
-					}
-				}
```

Comment:
I don't understand this - we won't delete the pods here but still check the amount of restarts in restartLimitExceeded?
the description of the PR is not that clear - what's the problem in restarting the pod and keeping the last one in failed state?

Reply:
i removed this part - the pod won't restart and we don't need to check anything regarding it.
`keeping the last one` ?

Reply:
> `keeping the last one` ?

yeah, I see in the description of the PR that "This change will allow to get the populator logs in case of failure in a easy way" but I don't understand why it's easier. let's say we try to populate a disk and we have three attempts and we fail in all of them - so the first populator pod would fail, we'll delete it, start a second populator pod that would also fail, we'll delete it, start a third populator pod that would also fail. why is it more complicated to get the logs from this pod compared to a single pod that we start and fail?

Reply:
if we delete the pod - it will be recreated, resetting the restartCount in that pod. that will cause us to be in an endless loop (restarted 3 times, recreate, restarted 3 times, recreate...).
it is easier because if the pod restarts or recreated, it depends on the phase you failed and the timing (if the pod just recreated/restarted you won't be able to get them).

Reply:
> it is easier because if the pod restarts or recreated, it depends on the phase you failed and the timing (if the pod just recreated/restarted you won't be able to get them).

right, but I don't see this as an issue since the migration should not be set with failure state at this point
if we manage to set the migration as failed only when the last attempt fails then we shouldn't care about the logs from previous attempt

the problem I see with this approach of changing the restart-policy of the populator pod is that currently CDI restarts them several times, now we drop this functionality, and I supposed that when we integrate our populator pods into CDI we'll also have several attempts..

Reply:
In CDI the annotation is set on the PVC, so there isn't really an issue with losing track of the count

Reply:
discussed offline with benny about it. @ahadas how about 3 re-creations and then to stop it, leaving the pod existing and telling the migration failed?
it means the number of creation need to be saved somewhere.

Reply:
yeah, that makes sense
I'd store it on the CR, assuming we can keep it this way also when the volume populators are integrated in CDI

Reply:
currently i created another commit. the pod will be recreated 3 times as agreed. i saved the annotation on the PVC. we can change it to the CR. i don't have strong opinion about it. as @bennyz said, CDI saves it on the PVC. but it depends how you wish it to be in CDI after integration.

Reply:
but isn't there a practical reason to store it on the PVC? I see the current code, it handles all populator pods - if we store the counter on the CR, we'll need to differentiate ovirt and openstack, no?

Reply:
yes, if we will store it on the CR we will need to differentiate, as it's unstructured.

Reply:
ack, so let's keep it on the PVC then